Source/WebCore/layout/formattingContexts/inline/InlineItemsBuilder.cpp

```diff
@@ -114,6 +114,58 @@ void InlineItemsBuilder::build(InlineItemPosition startPosition)
     };
     adjustInlineContentCacheWithNewInlineItems();
 
+    auto adjustElementBoundaryTextSpacings = [&] {
+        char32_t lastCharacterFromPreviousRun = 0;
+        unsigned lastCharacterDepth = 0;
+        unsigned currentCharacterDepth = 0;
+        ElementBoundaryTextSpacings spacings;
+        Vector<unsigned> inlineBoxStartIndexesOnInlineItemsList;
+        bool processElementBoundary = false;
+        auto& inlineItemList = inlineContentCache().inlineItems().content();
+        for (unsigned inlineItemIndex = 0; inlineItemIndex < inlineItemList.size(); ++inlineItemIndex) {
+            auto& inlineItem = inlineItemList[inlineItemIndex];
+            if (inlineItem.isInlineBoxStart()) {
+                inlineBoxStartIndexesOnInlineItemsList.append(inlineItemIndex);
+                ++currentCharacterDepth;
+                continue;
+            }
+            if (inlineItem.isInlineBoxEnd()) {
+                --currentCharacterDepth;
+                processElementBoundary = true;
+                continue;
+            }
+            auto* inlineTextItem = dynamicDowncast<InlineTextItem>(inlineItem);
+            if (!inlineTextItem)
+                continue;
+
+            auto start = inlineTextItem->start();
+            auto length = inlineTextItem->length();
+            auto& inlineTextBox = inlineTextItem->inlineTextBox();
+            if (!processElementBoundary || !lastCharacterFromPreviousRun) {
+                lastCharacterFromPreviousRun = inlineTextBox.content().characterAt(start + length - 1);
+                lastCharacterDepth = currentCharacterDepth;
+                processElementBoundary = false;
+                continue;
+            }
+
+            unsigned boundaryDepth = std::min(currentCharacterDepth, lastCharacterDepth);
+            unsigned boundaryIndex = inlineBoxStartIndexesOnInlineItemsList.size() - 1 - (currentCharacterDepth - boundaryDepth);
+            ASSERT(boundaryIndex >= 0);
+            const RenderStyle& boundaryStyle = inlineItemList[boundaryIndex].layoutBox().parent().style();
```

Comment:
"parent style" would be more informative as boundary is normally between 2 adjacent entries which makes it hard to reason about what style we are using here (also applies to 'boundary' here in general, e.g. not sure if boundary depth is descriptive enough)